lib/experimental/modules/scripts-modules.php

```diff
@@ -0,0 +1,183 @@
+		if ( str_starts_with( $field_name, 'editor' ) ) {
+			$asset_handle .= '-editor';
+		}
+		if ( str_starts_with( $field_name, 'view' ) ) {
+			$asset_handle .= '-view';
+		}
+		if ( $index > 0 ) {
+			$asset_handle .= '-' . ( $index + 1 );
+		}
+		return $asset_handle;
+	}
+
+	$field_mappings = array(
+		'viewModule' => 'view-module',
+	);
+	$asset_handle   = str_replace( '/', '-', $block_name ) .
+		'-' . $field_mappings[ $field_name ];
+	if ( $index > 0 ) {
+		$asset_handle .= '-' . ( $index + 1 );
+	}
+	return $asset_handle;
+}
+
+function gutenberg_register_view_module_ids_rest_field() {
+	register_rest_field(
+		'block-type',
+		'view_module_ids',
+		array(
+			'get_callback' => function ( $item ) {
+				$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $item['name'] );
+				if ( isset( $block_type->view_module_ids ) ) {
+					return $block_type->view_module_ids;
+				}
+				return array();
+			},
+		)
+	);
+}
+
+add_action( 'rest_api_init', 'gutenberg_register_view_module_ids_rest_field' );
+
+/**
+ * Registers the module if no module with that module identifier has already
+ * been registered.
+ *
+ * @param string            $module_identifier The identifier of the module. Should be unique. It will be used in the final import map.
+ * @param string            $src               Full URL of the module, or path of the script relative to the WordPress root directory.
+ * @param array             $dependencies      Optional. An array of module identifiers of the dependencies of this module. The dependencies can be strings or arrays. If they are arrays, they need an `id` key with the module identifier, and can contain a `type` key with either `static` or `dynamic`. By default, dependencies that don't contain a type are considered static.
+ * @param string|false|null $version           Optional. String specifying module version number. Defaults to false. It is added to the URL as a query string for cache busting purposes. If SCRIPT_DEBUG is true, the version is the current timestamp. If $version is set to false, the version number is the currently installed WordPress version. If $version is set to null, no version is added.
+ * @deprecated 17.4.2 gutenberg_register_module is deprecated. Use wp_register_module instead.
+ */
+function gutenberg_register_module( $module_id, $src, $deps = array(), $version = false ) {
+	_deprecated_function( __FUNCTION__, 'Gutenberg 17.4.2', 'wp_register_module' );
+	wp_modules()->register( $module_id, $src, $deps, $version );
+}
+
+/**
+ * Marks the module to be enqueued in the page.
+ *
+ * @param string $module_identifier The identifier of the module.
+ * @deprecated 17.4.2 gutenberg_enqueue_module is deprecated. Use wp_enqueue_module instead.
+ */
+function gutenberg_enqueue_module( $module_id, $src = '', $deps = array(), $version = false ) {
+	_deprecated_function( __FUNCTION__, 'Gutenberg 17.4.2', 'wp_enqueue_module' );
+	wp_modules()->enqueue( $module_id, $src, $deps, $version );
+}
```

Comment:
We don't need those because this was never supported in Gutenberg.

Also, shouldn't the Gutenberg version be `17.6` (the next one)?

```suggestion
function gutenberg_enqueue_module( $module_id  ) {
	_deprecated_function( __FUNCTION__, 'Gutenberg 17.6', 'wp_enqueue_module' );
	wp_modules()->enqueue( $module_id );
}
```

Reply:
17.5, I don't know what I was thinking about 😮‍💨

Reply:
If we do `wp_modules()->enqueue( $module_id );` it will require at least `$module_id` and `$src`.

Reply:
`$src` is optional in `wp_modules()->enqueue`

Reply:
I got confused with `wp_modules()->register` 🤦